### PR TITLE
Add resource routeros_ipv6_neighbor_discovery

### PR DESCRIPTION
**Pull request by jlpedrosa**
Implement #360  

* Add resource routeros_ipv6_neighbor_discovery. Tests, docs and example.
* Add  `___drop_val___` metadata. I allows to define special values per resource that when returned from mikrotik, it will be discarded. this is necessary due to the fact that Mikrotik returns literally **unspecified** for this resource when **some** fields are not sent.  It could have bee workarounded by setting the field as String and SupressDiffFunc. That has the huge drawback that the fields can't be Int, Bool .... so no validation is possible.

### CODE REVIEW

**Comment by jlpedrosa** [comment]
@vaerh I am a bit puzzled, tests are passing locally.

**Review by vaerh — APPROVED** [review]

**Comment by vaerh** [comment]
```
resource_ipv6_neighbor_discovery_test.go:14: Step 1/1 error: Check failed: Check 15/15 error: routeros_ipv6_neighbor_discovery.test: Attribute 'retransmit_interval' not found
```
I'll have a look now to see what's going on in the old version of ROS.

File: routeros/resource_ipv6_neighbor_discovery.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			Description: "Specify the router preference that is communicated to IPv6 hosts through router advertisements." +
+				"The preference value in the router advertisements enables IPv6 hosts to select a default router to reach a remote destination",
+			DiffSuppressFunc: AlwaysPresentNotUserProvided,
+		},
+		"retransmit_interval": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "The time between retransmitted Neighbor Solicitation messages." +
+				"Used by address resolution and the Neighbor Unreachability Detection algorithm (see Sections 7.2 and 7.3 of RFC 2461)",
+			DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```
Comment (vaerh) [inline routeros/resource_ipv6_neighbor_discovery.go:137]: I found the problem. It is contained in ```DiffSuppressFunc```.
JSON generation in the current implementation:
```
'retransmit-interval':''
'interface':'ether1'
'ra-delay':'3s'                                                                                                                                                                                                                                                                                                               
'ra-interval':'3m20s-10m'                                                                                                                                                                                                                                                                                                     'ra-lifetime':'30m'                                                                                                                                                                                                                                                                                                           
'reachable-time':'10m'                                                                                                                                                                                                                                                                                                        'ra-preference':'high'                                                                                                                                                                                                                                                                                                        
```
JSON with our nice and working ```AlwaysPresentNotUserProvided``` function
```
'retransmit-interval':'12m'
'interface':'ether1'
'ra-delay':'3s'
'ra-interval':'3m20s-10m'
'reachable-time':'10m'
'ra-preference':'high'
'ra-lifetime':'30m'
```
We should remove all such functions if they give such problems. If I remember correctly, we used them somewhere....

**Review by vaerh — CHANGES_REQUESTED** [review]

File: routeros/resource_ipv6_neighbor_discovery.go (hunk 1 above)
Comment (jlpedrosa) [reply routeros/resource_ipv6_neighbor_discovery.go:137]: Thank you very much... I left this here by accident while I was testing something. My apologies, I'll fix it soon.

**Review by vaerh — APPROVED** [review]

**Pull request merged**